prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/migrationtoolkit/GetAnalysisTask.java

```diff
@@ -0,0 +1,120 @@
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link GetAnalysisTask} gets analysis task from the MTA. This task returns the
+ * taskgroup details, including state of the analysis task and the url of the report if
+ * the task is in state succeeded.
+ */
+@Slf4j
+public class GetAnalysisTask extends BaseInfrastructureWorkFlowTask {
+
+	// The MTA application client to work with. If kept null then a client would be
+	// created using the context params serverURL and bearerToken during execution time.
+	// Not depending on the context params for the client details
+	// means flow authors can build the client object with their
+	// params and pass it is. This also prevents an invoker to pass the client
+	// details by ignoring them if a client already exists.
+	// This method is useful for testing as well.
+	protected MTATaskGroupClient mtaClient;
+
+	private URI serverUrl;
+
+	private final Consumer<Message> messageConsumer;
+
+	public GetAnalysisTask(URI serverURL, String bearerToken, Consumer<Message> messageConsumer) {
+		this.serverUrl = serverURL;
+		this.mtaClient = new MTAClient(serverURL, bearerToken);
+		this.messageConsumer = messageConsumer;
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key("taskGroupID").type(WorkParameterType.NUMBER).optional(false)
+						.description("The application name as presented in the application hub").build(),
+				WorkParameter.builder().key("serverURL").type(WorkParameterType.TEXT).optional(true).description(
+						"Base URL of the MTA instance - e.g https://mta-openshift-mta.app.clustername.clusterdomain")
+						.build(),
+				WorkParameter.builder().key("bearerToken").type(WorkParameterType.TEXT).optional(true)
+						.description("Bearer token to authenticate server requests").build());
+	}
+
+	/**
+	 * @param workContext optional context values: serverURL, and bearerToken for the
+	 * mtaClient.
+	 */
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		if (mtaClient == null) {
+			serverUrl = URI.create(getOptionalParameterValue(workContext, "serverURL", null));
+			var bearerToken = getOptionalParameterValue(workContext, "bearerToken", null);
+			if (serverUrl == null) {
+				log.error(
+						"serverURL is empty. Either pass it while creating the instance of the task or in the context");
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext);
+			}
+			mtaClient = new MTAClient(serverUrl, bearerToken);
+		}
+
+		int taskGroupID;
+		try {
+			taskGroupID = Integer.parseInt(getRequiredParameterValue(workContext, "taskGroupID"));
+		}
+		catch (MissingParameterException | NumberFormatException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
```

Comment:
Funny, this feels like the canDoAction / validateCommand we used to have before the commandExecute.
I wonder if it is worth to explore (not in this PR scope) the extension of a single execute() to more than just one (e.g. validateContext prior to execution, in which each can add the needed logic to verify the context matches the expectation)